web-animations-2/Overview.bs

```diff
@@ -213,6 +213,9 @@ follows:
     abort this procedure.
 1.  Let |previous play state| be |animation|'s [=play state=].
 1.  Let |previous current time| be the |animation|'s [=current time=].
+1.  Let |previous progress| = <code>|previous current time| /
+    [=end time=]</code> if |previous current time| is resolved and
```

Comment:
What if end time is 0? I think we need to have a special case in order to avoid division by 0.

What if end time is infinity? This will break preserving the current progress for any infinite iteration animation when going between document timelines I think.

See https://drafts.csswg.org/web-animations-1/#end-time

Maybe we should only use the previous progress if we are switching from a progress based to time based or time based to progress based animation, as these are the only two cases where we can't directly assign the progress from the other timeline.

Reply:
Good catch. Missed a step where previous progress should be set to zero if end time is zero.  The case of end time being infinite should be fine since we don't preserve current time if transitioning between monotonic timelines.  It will simply mean that previous progress will be zero in this case.  Previous progress is only used in two special cases where we want to avoid an animation jump: monotonic to non-monotonic while paused, and non-monotonic to monotonic.

Reply:
Expanding on my previous comment.  We will see a jump when transitioning from a monotonic to non-monotonic timeline on a paused animation if the previous end time was infinite.  This seems unavoidable.  In fact, we explicitly disallow a progress based animation with zero iteration duration.  This case trips a DOM error if the animation is initially set up with a progress based timeline.  Unclear if it will trip and error downstrream after setting the timeline.  Would like to follow up in a separate PR if additional handling is needed for this case.